Pump 1.13.2 RNs.mdx (#1066)
TykTechnologies/tyk-docs: developer-support/release-notes/pump.mdx

## 1.13 Release Notes

### 1.13.2 Release Notes

#### Release Date 16 December 2025

#### Release Highlights

In this patch release, we've resolved SQL Pump schema migration issues for sharded tables. For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.13.2).

#### Breaking Changes

> 

Reviewer: Can we add a "no breaking changes" note? It should point out that there's potential for a start-up delay if schema migration is enabled (directing them to the change log entry).

## 1.13 Release Notes

### 1.13.2 Release Notes

#### Release Date 17 December 2025

#### Release Highlights

In this patch release, we've resolved SQL Pump schema migration issues for sharded tables. For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.13.2).

#### Breaking Changes

> No breaking changes. However, please note that first startup may be slower due to database schema migration. See changelog for details.

Reviewer: ```suggestion
No breaking changes. However, please note that first startup may be slower due to database schema migration if the new `migrate_sharded_tables` configuration is set. See changelog for details.
```

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

There are no deprecations in this release.
For further information, please check the [changelog](#Changelog-v1.13.2) below.

#### Upgrade instructions

For users currently on v1.13.1, we strongly recommend promptly upgrading to the latest release. If you are working with an older version (lower major), it is advisable to bypass version 1.13.0 and proceed directly to this latest patch release.
<br/>
Go to the [Upgrading Tyk](#upgrading-tyk) section for detailed upgrade Instructions.

#### Downloads
- [Docker Image v1.13.2](https://hub.docker.com/r/tykio/tyk-pump-docker-pub/tags?page=&page_size=&ordering=&name=v1.13.2)
  - ```bash
    docker pull tykio/tyk-pump-docker-pub:v1.13.2
    ```
- Source code tarball for OSS - [GH Tyk Pump Repo](https://github.com/TykTechnologies/tyk-pump/releases/tag/v1.13.2)

#### Changelog
<a id="Changelog-v1.13.2"></a>

##### Changed

<AccordionGroup>

> <Accordion title='Storage library v1.3.0 Update Across All Tyk Components'>

Reviewer 2: ```suggestion
<Accordion title='Update Storage library to v1.3.0'>
```

Updated the storage library to v1.3.0 to include MongoDB connection fixes and foundational SQL support. All components now benefit from improved connection management and enhanced storage layer reliability.
</Accordion>

> 

Reviewer 2: ```suggestion
</AccordionGroup>

```

## 1.13 Release Notes

### 1.13.2 Release Notes

#### Release Date xx December 2025

#### Release Highlights

> In this patch release, we enhanced storage reliability with updated storage library v1.3.0 and resolved SQL Pump schema migration issues for sharded tables. For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.13.2).

Reviewer: ```suggestion
In this patch release, we have resolved SQL Pump schema migration issues for sharded tables. For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.13.2).
```

Reviewer 2: This change has been incorporated manually.

## 1.13 Release Notes

### 1.13.2 Release Notes

> #### Release Date xx December 2025

Reviewer 2: TODO: Change this while merging

    ```
- Source code tarball for OSS - [GH Tyk Pump Repo](https://github.com/TykTechnologies/tyk-pump/releases/tag/v1.13.2)

#### Changelog
<a id="Changelog-v1.13.2"></a>

##### Changed

<AccordionGroup>
<Accordion title='Storage library v1.3.0 Update Across All Tyk Components'>
Updated the storage library to v1.3.0 across all Tyk components (Gateway, Dashboard, MDCB, Pump, Sync, and Portal) to include MongoDB connection fixes and foundational SQL support. All components now benefit from improved connection management and enhanced storage layer reliability.
</Accordion>

##### Fixed

<AccordionGroup>

<Accordion title='Fixed SQL Pump table sharding schema migration issues'>
Fixed an issue when using table sharding with the SQL pumps (creating new tables for each day), where the Pump would generate error messages in the application logs due to missing columns in the tables. This would only occur when upgrading to a newer version of Pump that added columns (e.g., to Pump 1.11.0, which added some GraphQL-related columns). Now, when Pump is upgraded, the current day's table will automatically be updated to include the new columns, so the error messages will not be generated.

> We have added a new configuration option `migrate_sharded_tables`, which will update all pre-existing tables in the SQL database to match the latest schema. This will be a one-time operation when the Pump starts up, but could take some time to complete if there is a large history in the database, so it has been made an optional activity via this configuration option.

Reviewer 2: We need a reference to the documentation page for `migrate_sharded_tables`.

Author: This will be added automatically to the https://tyk.io/docs/tyk-pump/tyk-pump-configuration/tyk-pump-environment-variables right?
Do you think we need to document it anywhere else?

Reviewer 2: Yes, you are correct. But we need to add a reference to that specific section on the page. Once the PR for the pump env is created, we can add the reference URL.

Example:
`[migrate_sharded_tables](/tyk-pump/tyk-pump-configuration/tyk-pump-environment-variables#migrated_shar)`
